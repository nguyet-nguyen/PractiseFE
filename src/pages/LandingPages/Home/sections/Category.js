import React, { useState } from "react";
import { Link } from "react-router-dom";
import _ from "lodash";

const Category = ({categoryList}) => {

    return (
        <div id="categoryHome">
            <h3 className="text-color uppercase text-center font-semibold md:text-3xl text-xl">Product's Categories</h3>
            <p className="text-color text-center mb-4 text-base product-desc">All products are warranted for 12 months</p>
            <div className="flex flex-wrap md:justify-center justify-start">
                {categoryList && categoryList.map((category, index) =>
                    (index > 0) ?
                        <div key={category.id} className={`flex md:flex-wrap md:w-1/5 w-1/2 flex-col category category-${index}`}>
                            <div className="w-full p-2 md:p-4">
                                <div className="text-center">
                                    <Link to={`/all-items/${category.id}`}>
                                        <div className="category-image relative overflow-hidden
                                         mx-auto md:w-56 md:h-56 w-32 h-32 md:mb-4 mb-2 ">
                                            <img
                                                src={category.image}
                                                className="rounded-full w-full h-full image block"
                                                alt="Avatar"
                                            />
                                            <div className="overlay bg-gray-800 rounded-full opacity-0 inset-0 absolute h-full w-full"> </div>
                                        </div>
                                        <h5 className="md:text-xl text-base font-semibold leading-tight text-color hover:text-amber-700">{category.name}</h5>
                                    </Link>
                                </div>
                            </div>
                        </div> : ''
                )}
            </div>
        </div>



    );
}

export default Category;