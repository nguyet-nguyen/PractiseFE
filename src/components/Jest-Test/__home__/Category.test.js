import React from "react";
import ReactDOM from "react-dom";
import { act } from 'react-dom/test-utils';
import Category from "pages/LandingPages/Home/sections/Category";
import {
    BrowserRouter,
} from "react-router-dom";
describe("Unit Test category homepage Component render", () => {
    let container;

    beforeEach(() => {
        container = document.createElement('div');
        document.body.appendChild(container);
    })

    afterEach(() => {
        document.body.removeChild(container);
        container = null;
    })

    it("SnapShot case 1", () => {
        const categoryList = []
        act(async () => {
            ReactDOM.render(
                <BrowserRouter>
                    <Category />
                </BrowserRouter>
                , container)
        });
        let categoryHome = container.querySelector('#categoryHome');
        expect(categoryHome.querySelector('.product-desc').textContent).toBe('All products are warranted for 12 months');
        expect(categoryHome).toMatchSnapshot();
    });

    it("SnapShot case 2", async () => {
        const categoryList = [{
            id: 2,
            name: "Pan",
            image: "https://bizweb.dktcdn.net/thumb/large/100/419/519/products/quanonglung.jpg?v=1614139227953",
        }];

        act(async () => {
            ReactDOM.render(
                <BrowserRouter>
                    <Category />
                </BrowserRouter>
                , container)
        });

        let categoryHome = container.querySelector('#categoryHome');
        // expect(container.querySelector(".categoryName").textContent).toBe('Pan');
        expect(categoryHome).toMatchSnapshot();

    })

})