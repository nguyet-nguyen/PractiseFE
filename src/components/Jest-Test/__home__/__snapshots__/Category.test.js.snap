// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unit Test category homepage Component render SnapShot case 1 1`] = `
<section
  class="overflow-hidden"
  id="categoryHome"
>
  <div
    class="container px-5 py-2 mx-auto md:py-12 md:px-32"
  >
    <h3
      class="text-color uppercase text-center font-semibold md:text-3xl text-xl"
    >
      Product's Categories
    </h3>
    <p
      class="text-color text-center mb-4 text-base product-desc"
    >
      All products are warranted for 12 months
    </p>
    <div
      class="flex flex-wrap justify-center"
    />
  </div>
</section>
`;

exports[`Unit Test category homepage Component render SnapShot case 2 1`] = `
<section
  class="overflow-hidden"
  id="categoryHome"
>
  <div
    class="container px-5 py-2 mx-auto md:py-12 md:px-32"
  >
    <h3
      class="text-color uppercase text-center font-semibold md:text-3xl text-xl"
    >
      Product's Categories
    </h3>
    <p
      class="text-color text-center mb-4 text-base product-desc"
    >
      All products are warranted for 12 months
    </p>
    <div
      class="flex flex-wrap justify-center"
    />
  </div>
</section>
`;
